client/src/store/index.js

```diff
@@ -40,34 +46,29 @@ export function createStore() {
             datasetPathDestination: datasetPathDestinationStore,
             datasetExtFiles: datasetExtFilesStore,
             invocations: invocationStore,
-            user: userStore,
-            config: configStore,
             workflows: workflowStore,
             datasets: datasetsStore,
             informationStore: jobStore,
-            betaHistory: betaHistoryStore,
         },
     };
 
-    // Initialize state
-
+    // Initializes global store elements.
+    // Vuex can't lazy-load, so we much manually insert global data at some point and we currently
+    // don't have a root App component container (which is traditionally used for such operations).
```

Comment:
In a single page app there is usually only one mount or starting point for the app. At that point you would traditionally populate any global data (ajax call to get current user, load application configs etc.), and that would be triggered on some a created() hook in a root <app> component or something.

But, Galaxy is not yet that kind of app and we are mounting in multiple places from within backbone by calling individualized mount functions which will fire up a new Vue instance. Although they will share a global Vuex, the issue becomes "when do we initialize that Vuex instance?"

Vuex is by design a simple synchronous state manager. If you want to get data out,  at the end of the day this means that somebody, somewhere has to fire an action to populate the state, and that is what a provider can give us. because it is a component with lifecycle hooks.

When I am done fixing the Upload modal, or any other Vue component that is mounted independently they would look something like this:

```html
<AppConfig v-slot="config">
   <UploadModal v-if="config" :thing-i-need="config.thing" :other-relevent-setting="config.otherThing" />
</AppConfig>
```
Here AppConfig is a provider that, more than likely is just going to pass out a slot prop peeled directly from the Vuex state, but it is giving us a lifecycle hook which guarantees that we have the opportunity to load the config from the server if it's not yet available. We can check to make sure we don't do it more times than we wanted, but we have the necessary async hook to do so.

Does that make my plans clear?

Reply:
Awesome, can you just copy-paste that into a planning issue?

Reply:
https://github.com/galaxyproject/galaxy/issues/11077

Reply:
Thanks!